File: app/src/main/java/com/organizer/layouts/HorizontalScrollLayout.java
package com.organizer.layouts;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.view.ViewGroup;
import android.widget.HorizontalScrollView;
import android.widget.LinearLayout;
import com.organizer.MainActivity;
import com.organizer.layouts.todo.ToDoLayout;

public class HorizontalScrollLayout extends HorizontalScrollView
{
    private boolean scrollable = true;
    private Context context;
    LinearLayout linearLayout;
    ToDoLayout toDoLayout;
    
    public boolean isScrollable()
    {
        return scrollable;
    }
    
    public void setScrollable(boolean scrollable)
    {
        this.scrollable = scrollable;
    }
    
    public HorizontalScrollLayout(Context context, AttributeSet attrs, int defStyle)
    {
        super(context, attrs, defStyle);
        this.context = context;
        init();
    }
    
    public HorizontalScrollLayout(Context context, AttributeSet attrs)
    {
        super(context, attrs);
        this.context = context;
        init();
    }
    
    public HorizontalScrollLayout(Context context)
    {
        super(context);
        this.context = context;
        init();
    }
    
    private void init()
    {
        setHorizontalScrollBarEnabled(false);
        setOverScrollMode(OVER_SCROLL_NEVER);
        setLayoutParams(new ViewGroup.LayoutParams(
            ViewGroup.LayoutParams.WRAP_CONTENT,
            ViewGroup.LayoutParams.MATCH_PARENT));
        
        linearLayout = new LinearLayout(context);
        linearLayout.setLayoutParams(new ViewGroup.LayoutParams(
            ViewGroup.LayoutParams.WRAP_CONTENT,
            ViewGroup.LayoutParams.MATCH_PARENT));
        addView(linearLayout);
    }
    
    public void addContentView(ToDoLayout contentView)
    {
        toDoLayout = contentView;
        linearLayout.addView(contentView);
    }
    
    @Override
    public boolean onInterceptTouchEvent(MotionEvent ev)
    {
        if (isScrollable())
        {
            return super.onInterceptTouchEvent(ev);
        }
        else
        {
            return false;
        }
    }
    
    @Override
    public boolean onTouchEvent(MotionEvent ev)
    {
        if (isScrollable())
        {
            super.performClick();
            return super.onTouchEvent(ev);
        }
        else
        {
            return false;
        }
    }
    
    @Override
    public boolean performClick()
    {
        return super.performClick();
    }
    
    @Override
    protected void onScrollChanged(int left, int top, int oldLeft, int oldTop)
    {
        super.onScrollChanged(left, top, oldLeft, oldTop);
        int width = toDoLayout.getMaxWidth() - MainActivity.getDisplayMetricsController().getScreenWidth();
        if (left > width)
        {
            scrollTo(width, 0);
        }
    }
}

File: app/src/main/kotlin/com/organizer/layouts/HorizontalScrollLayout.kt
package com.organizer.layouts

import android.annotation.SuppressLint
import android.view.MotionEvent
import android.view.ViewGroup
import android.widget.HorizontalScrollView
import android.widget.LinearLayout
import com.organizer.*
import com.organizer.MainActivity.Companion.inject
import com.organizer.MainActivity.Companion.injectNow
import com.organizer.layouts.todo.ToDoLayout

class HorizontalScrollLayout : HorizontalScrollView(injectNow()) {

    private val displayMetricsController: DisplayMetricsController by inject()

    var isScrollable = true
    private val linearLayout = LinearLayout(context)
    private var toDoLayout: ToDoLayout? = null

    init {
        isHorizontalScrollBarEnabled = false
        overScrollMode = OVER_SCROLL_NEVER
        layoutParams = ViewGroup.LayoutParams(
            ViewGroup.LayoutParams.WRAP_CONTENT,
            ViewGroup.LayoutParams.MATCH_PARENT
        )

        addView(linearLayout.apply {
            layoutParams = ViewGroup.LayoutParams(
                ViewGroup.LayoutParams.WRAP_CONTENT,
                ViewGroup.LayoutParams.MATCH_PARENT
            )
        })
    }

    fun addContentView(contentView: ToDoLayout) {
        toDoLayout = contentView
        linearLayout.addView(contentView)
    }

    override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
        return if (isScrollable) {
            super.onInterceptTouchEvent(ev)
        } else {
            false
        }
    }

    @SuppressLint("ClickableViewAccessibility")
    override fun onTouchEvent(ev: MotionEvent): Boolean {
        return if (isScrollable) {
            super.performClick()
            super.onTouchEvent(ev)
        } else {
            false
        }
    }

    override fun onScrollChanged(left: Int, top: Int, oldLeft: Int, oldTop: Int) {
        super.onScrollChanged(left, top, oldLeft, oldTop)
        toDoLayout?.let {
            val width = it.maxWidth - displayMetricsController.screenWidth
            if (left > width) {
                scrollTo(width, 0)
            }
        }
    }
}
